rcldotnet/RCLdotnet.cs

```diff
@@ -198,9 +221,36 @@ public class RCLdotnet {
 
     public static Node CreateNode (string nodeName, string nodeNamespace) {
       IntPtr nodeHandle = IntPtr.Zero;
-      int ret = RCLdotnetDelegates.native_rcl_create_node_handle (ref nodeHandle, nodeName, nodeNamespace);
-      Node node = new Node (nodeHandle);
-      return node;
+      RMWRet ret = (RMWRet)RCLdotnetDelegates.native_rcl_create_node_handle (ref nodeHandle, nodeName, nodeNamespace);
+
+      switch (ret) {
```

Comment:
This can be simplified greatly by just first checking if the name is valid (i.e. adding a `native_rmw_validate_node_name`) and then calling `native_rcl_create_node_handle`.

Reply:
Split into two functions. Now code is more clear

Reply:
It's an improvement, but it could still be a bit more readable. There's a lot of duplicated code, another way to write this is to check if ret is RCLRet.Ok with an if statement, and then do the switch.

Reply:
Refactored